source/_components/pcal9535a.markdown

```diff
@@ -0,0 +1,130 @@
+
+One of the use cases is [Seeed studio Raspberry Pi Relay Board](http://wiki.seeedstudio.com/Raspberry_Pi_Relay_Board_v1.0/).
+
+For more details about the PCAL9535A I2C I/O port expander you can find its datasheet here: [PCAL9535A](https://www.nxp.com/docs/en/data-sheet/PCAL9535A.pdf).
+
+## Binary Sensor
+
+The `pcal9535a` binary sensor platform allows you to read sensor values from the I/O pins of your [PCAL9535A I2C I/O expander](https://www.nxp.com/products/interfaces/ic-spi-serial-interface-devices/ic-general-purpose-i-o/low-voltage-16-bit-ic-bus-i-o-port-with-interrupt-and-agile-i-o:PCAL9535A).
+
+The pin numbers are from 0 to 15 where: 0-7 correspond to port P0 (P0_0 - P0_7) and 8-15 to port P1 (P1_0 - P1_7).
+
+### Configuration
+
+To use the I/O pins of an pcal9535a connected to an I2C bus of your Raspberry Pi as binary sensors, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+binary_sensor:
+  - platform: pcal9535a
+    i2c_address: 0x20
+    pins:
+      0: PIR Office
+      1: PIR Bedroom
+```
+
+{% configuration %}
+i2c_bus:
+  description: i2c bus number of PCAL9535A chip.
+  required: false
+  type: integer
+  default: 1
+i2c_address:
+  description: i2c address of PCAL9535A chip.
+  required: false
+  type: integer
+  default: "`0x20`"
+pins:
+  description: List of used pins.
+  required: true
+  type: map
+  keys:
+    "pin: name":
+      description: The pin numbers (from 0 to 15) and corresponding names.
+      required: true
+      type: [integer, string]
+scan_interval:
+  description: Interval to scan for sensor state changes in seconds.
+  required: false
+  type: integer
+  default: 15
+invert_logic:
+  description: If `true`, inverts the output logic to ACTIVE LOW.
+  required: false
+  type: boolean
+  default: "`false` (ACTIVE HIGH)"
+pull_mode:
+  description: >
+    Type of internal pull resistor to use.
+    Options are `UP` - pull-up resistor, `DOWN` - pull-down resistor, `DISABLED` - resistors disconnected.
+  required: false
+  type: string
+  default: "`DISABLED`"
+{% endconfiguration %}
+
+## Switch
+
+The `pcal9535a` switch platform allows you to write to the I/O pins of your [PCAL9535A I2C I/O expander](https://www.nxp.com/products/interfaces/ic-spi-serial-interface-devices/ic-general-purpose-i-o/low-voltage-16-bit-ic-bus-i-o-port-with-interrupt-and-agile-i-o:PCAL9535A).
+
+The pin numbers are from 0 to 15 where: 0-7 correspond to port P0 (P0_0 - P0_7) and 8-15 to port P1 (P1_0 - P1_7).
+
+### Configuration
+
+To use the I/O pins of an pcal9535a connected to an I2C bus of your Raspberry Pi as switches, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+switch:
+  - platform: pcal9535a
+    i2c_address: 0x20
```

Comment:
```suggestion
```

Reply:
I made this 'suggestion' to myself just in order to apply all changes in one batch in a single commit.